pkg/git/git_test.go

```diff
@@ -206,6 +228,27 @@ func TestFetch(t *testing.T) {
 				t.Errorf("Fetch() error = %v, wantErr %v", err, tt.wantErr)
 			}
 
+			if tt.spec.SparseCheckoutDirectories != "" {
+				dirPatterns := strings.Split(tt.spec.SparseCheckoutDirectories, tt.spec.SparseCheckoutDelimiter)
+
+				sparseFile, err := os.Open(".git/info/sparse-checkout")
+				if err != nil {
+					t.Fatal("Unable to read sparse-checkout file")
+				}
+				defer sparseFile.Close()
+
+				var sparsePatterns []string
+
+				scanner := bufio.NewScanner(sparseFile)
+				for scanner.Scan() {
+					sparsePatterns = append(sparsePatterns, scanner.Text())
+				}
+
+				if !reflect.DeepEqual(dirPatterns, sparsePatterns) {
```

Comment:
what about using cmp.Diff, which is a library used thought-out all the tests in pipeline ? correct me if I am wrong but it does the same thing ?

Reply:
cmp.Diff will work here. I will make that change!